### PR TITLE
Moved account to BaseEth

### LINKED ISSUE #2580: Add 'account' to AsyncEth?
Hey 👋 Is there a reason `account` is not available in `AsyncEth`?

So far, I had to create a non-async w3 object to sign transactions. However, I just found out that when I subclass AsyncEth like this
```python
class CustomAsyncEth(AsyncEth):
    account = Account()
```
signing transactions works just fine.

### PR DESCRIPTION
**Pull request by dbfreem**
### What was wrong?

Make account accessible in AsyncEth

closes #2580 

### How was it fixed?

Move Eth.account to BaseEth so that it can be used in AsyncEth.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### CODE REVIEW

**Review by kclowes — COMMENTED** [review]
Thanks @dbfreem! Before this gets merged, we'll need to add a few tests. If you want to/have time to add them, that would be great! Otherwise someone from the team can pick it up! 

You should be able to pretty much wholesale copy/paste `tests/core/eth-module/test_async_accounts.py`, but where you have your `async_w3` fixture defined, you'll need to pass in the `AsyncEth` module. So instead of [this line](https://github.com/ethereum/web3.py/blob/master/tests/core/eth-module/test_accounts.py#L88) it should look something like: 

`return Web3(AsyncEthereumTesterProvider(), modules={"eth", (AsyncEth,)})`

**Comment by dbfreem** [comment]
So, I did some thinking about the test here. Two things I thought of.

1.  Most of these test are testing the functionality of `eth_account.Account`. It could be argued that these test don't really belong in web3.py but actually in eth_account. They are actually probably already covered there.
2. If we do need test for this, shouldn't the test be only on BaseEth instead of duplicating them between AsyncEth and Eth? I could change the above referenced test to be only one set of test against `BaseEth.account`.

Just my thoughts so feel free to say "yes, we want both the AsynEth and Eth test in web3.py for account". :) haha.

**Comment by kclowes** [comment]
Thanks for the comments!

> 1. Most of these test are testing the functionality of eth_account.Account. It could be argued that these test don't really belong in web3.py but actually in eth_account. They are actually probably already covered there.

They are covered over there as well, but I think it's good to make sure we can access the most important methods through `w3.eth.account`, rather than just on the `Account` class. For example, when the HDWallet feature was implemented, it was working great in `eth-account`, but we had to do some tweaking to get it to work in web3. 

> 2. If we do need test for this, shouldn't the test be only on BaseEth instead of duplicating them between AsyncEth and Eth? I could change the above referenced test to be only one set of test against BaseEth.account.

That's a good idea. I think it would be good to have at least one "happy path" test for each `Eth` and `AsyncEth` to make sure we can access `account` through each of those classes to prevent any regressions though.

**Comment by dbfreem** [comment]
@kclowes  take a look at this and let me know what you think

**Review by fselmo — COMMENTED** [review]
I added a comment just to make sure we have decent assumptions for the testing around this. It would be good to add this feature to the async docs too but I can do that before merging if you'd like. Just let me know.

Thanks for the PR 👌

File: newsfragments/2580.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Moved `Eth.account`` to BaseEth.account so that it could be use in `AsyncEth``
```
[the remark this answers is not in the document]
Comment (dbfreem) [reply newsfragments/2580.feature.rst]: Good Catch! This is fixed now.

File: tests/core/eth-module/test_accounts.py [hunk 2]
```diff
@@ -88,6 +95,21 @@ def w3():
...
+def async_w3():
+    return Web3(AsyncEthereumTesterProvider(), modules={"eth": [AsyncEth]})
+
+
+def test_blocking_eth_account(w3):
+    account = w3.eth.account.create()
+    assert account is not None
+
+
+def test_async_eth_account(async_w3):
```
[the remark this answers is not in the document]
Comment (dbfreem) [reply tests/core/eth-module/test_accounts.py:115]: I copied these test into tests/core/eth-module/test-accounts.py

**Comment by dbfreem** [comment]
@fselmo I made the suggested changes. Copied your test into test_account.py, and made the doc changes.

**Comment by fselmo** [comment]
> @fselmo I made the suggested changes. Copied your test into test_account.py, and made the doc changes.

lgtm! I just organized async at the bottom so it's a bit easier to read. Ty 👍🏼... I'll merge as soon as tests pass

**Pull request merged**